features/core.feature

```diff
@@ -20,6 +20,20 @@ Feature: Basic reading and writing to a journal
         When we run "jrnl -n 1"
         Then the output should contain "2013-07-23 09:00 A cold and stormy day."
 
+    Scenario: Writing an empty entry from the editor
+        Given we use the config "editor.yaml"
+        When we open the editor and enter ""
+        Then we should see the message "[Nothing saved to file]"
+
+    Scenario: Writing an empty entry from the command line
+        Given we use the config "basic.yaml"
+        When we run "jrnl ''"
```

Comment:
```suggestion
        When we run "jrnl" and enter ""
```

Reply:
That causes the test to hang until `Ctrl-c` or `Ctrl-D` is entered. It does make the test pass when it is, however.

Reply:
I see, the mocking doesn't work with an empty string currently

Reply:
I updated it to work with an empty string. I'm not sure why `sys.stdin.read()` isn't being mocked here.

<img width="897" alt="image" src="https://user-images.githubusercontent.com/20600565/69058319-bc6ae580-0a13-11ea-865c-6f46e8b34a1d.png">